airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -93,15 +94,30 @@ def inner_wrapper(self: "DataflowHook", *args, **kwargs) -> RT:
 class DataflowJobStatus:
     """
     Helper class with Dataflow job statuses.
+    Reference: https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#Job.JobState
     """
-    JOB_STATE_DONE = "JOB_STATE_DONE"
+    JOB_STATE_UNKNOWN = "JOB_STATE_UNKNOWN"
```

Comment:
nit: for var names is `JOB_STATE_` prefix really necessary on all these?
IMO slightly more readable to drop it and this causes "stutter" `DataflowJobStatus.JOB_STATE_xxx`.

A forward looking thought (though not backwards compatible so not immediate suggestion) in python 3.8+ this set could be more concise  with walrus operator e.g.
```python
FAILED_END_STATES = {
    (FAILED := "JOB_STATE_FAILED"),
    (CANCELLED := "JOB_STATE_CANCELLED"),
    (STOPPED := "JOB_STATE_STOPPED")
}
```

Reply:
created separate issue for stutter: https://github.com/apache/airflow/issues/11205